packages/yaml/src/index.ts

```diff
@@ -0,0 +1,129 @@
+import { type Service } from '@volar/language-service';
+import { type TextDocument } from 'vscode-languageserver-textdocument';
+import { type LanguageSettings, type LanguageService } from 'yaml-language-server';
+import { getLanguageService } from 'yaml-language-server/lib/umd/languageservice/yamlLanguageService.js';
+
+export interface Provide {
+	'yaml/languageService': () => LanguageService;
+}
+
+function isYaml(document: TextDocument): boolean {
+	return document.languageId === 'yaml';
+}
+
+function noop(): undefined { }
+
+/**
+ * Create a Volar language service for YAML documents.
+ */
+export default (settings: LanguageSettings): Service<Provide> => {
+	return (context): ReturnType<Service<Provide>> => {
+
+		if (!context) {
+			return { triggerCharacters: [' ', ':'] } as any;
+		}
+
+		const ls = getLanguageService({
+			schemaRequestService: async (uri) => await context.env.fs?.readFile(uri) ?? '',
+			telemetry: {
+				send: noop,
+				sendError: noop,
+				sendTrack: noop
+			},
+			// @ts-expect-error https://github.com/redhat-developer/yaml-language-server/pull/910
+			clientCapabilities: context?.env?.clientCapabilities,
+			workspaceContext: {
+				resolveRelativePath(relativePath, resource) {
+					return String(new URL(relativePath, resource));
```

Comment:
I found this is simpler than json service. That code of json service (https://github.com/volarjs/services/blob/a2eee58c30eabb245b27615be3056c4f3b6f4088/packages/json/src/index.ts#L21-L32) is modified from vscode implementation. I think this implementation is also works, let's keep it.